java-frontend/src/main/java/org/sonar/java/resolve/TypeAndReferenceSolver.java

```diff
@@ -892,6 +933,7 @@ public void visitArrayType(ArrayTypeTree tree) {
 
   @Override
   public void visitTypeCast(TypeCastTree tree) {
+    // FIXME SONARJAVA-2724: should be the intersection of the type with its bounds
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Take the required action to fix the issue indicated by this comment. ([squid:S1134](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1134))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.java%3Ajava&pullRequest=1992&issues=AWK1qcFnEyLEh1Ch10th&open=AWK1qcFnEyLEh1Ch10th)

[ISSUE_KEY]: AWK1qcFnEyLEh1Ch10th